kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/InCommitTimestampSuite.scala

```diff
@@ -0,0 +1,179 @@
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = TableImpl.forPath(engine, tablePath)
+
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = true, IN_COMMIT_TIMESTAMPS_ENABLED, "true", true)
+
+      val ver0Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver0Snapshot, TableConfig.IN_COMMIT_TIMESTAMP_ENABLEMENT_TIMESTAMP,
+        Optional.empty())
+      assertMetadataProp(ver0Snapshot, TableConfig.IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION,
+        Optional.empty())
+    }
+  }
+
+  test("Enablement tracking works when ICT is enabled post commit 0") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = TableImpl.forPath(engine, tablePath)
+      val txnBuilder = table.createTransactionBuilder(engine, testEngineInfo, CREATE_TABLE)
+
+      val txn = txnBuilder
+        .withSchema(engine, testSchema)
+        .build(engine)
+
+      txn.commit(engine, emptyIterable())
+
+      appendData(
+        engine,
+        tablePath,
+        schema = testSchema,
+        partCols = Seq.empty,
+        data = Seq(Map.empty[String, Literal] -> dataBatches1),
+        tableProperties = Map(IN_COMMIT_TIMESTAMPS_ENABLED.getKey -> "true"))
+
+      val ver1Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver1Snapshot, TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED, true)
+      assertMetadataProp(
+        ver1Snapshot,
+        TableConfig.IN_COMMIT_TIMESTAMP_ENABLEMENT_TIMESTAMP,
+        getInCommitTimestamp(engine, table, version = 1))
+      assertMetadataProp(
+        ver1Snapshot,
+        TableConfig.IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION, Optional.of(1L))
+
+      appendData(
+        engine,
+        tablePath,
+        schema = testSchema,
+        partCols = Seq.empty,
+        data = Seq(Map.empty[String, Literal] -> dataBatches2))
+
+      val ver2Snapshot = table.getLatestSnapshot(engine).asInstanceOf[SnapshotImpl]
+      assertMetadataProp(ver2Snapshot, TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED, true)
+      assertMetadataProp(
+        ver2Snapshot,
+        TableConfig.IN_COMMIT_TIMESTAMP_ENABLEMENT_TIMESTAMP,
+        getInCommitTimestamp(engine, table, version = 1))
+      assertMetadataProp(
+        ver2Snapshot,
+        TableConfig.IN_COMMIT_TIMESTAMP_ENABLEMENT_VERSION, Optional.of(1L))
+    }
+  }
+
+  test("Update the protocol unless required") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = true, IN_COMMIT_TIMESTAMPS_ENABLED, "true", true)
+      val protocol = getProtocolActionFromCommit(engine, table, 0)
+      assert(protocol.isPresent)
+      assert(protocol.get.getWriterFeatures.contains("inCommitTimestamp-preview"))
+
+      setTablePropAndVerify(
+        engine, tablePath, isNewTable = false, IN_COMMIT_TIMESTAMPS_ENABLED, "false", false)
```

Comment:
How is this working? Shouldn't we throw error if the ICT is disabled?

Reply:
@dhruvarya-db how does Delta-Spark handles when the ICT is disabled?

Reply:
Disabling ICT by setting the enablement property to `false` is possible even in Delta-Spark. Do you see any problems with this? Disabling ICT this way does not remove it from the protocol though.